virtualizarr/writers/icechunk.py

```diff
@@ -118,6 +117,82 @@ def dataset_to_icechunk(
     )
 
 
+def datatree_to_icechunk(
+    dt: xr.DataTree,
+    store: "IcechunkStore",
+    *,
+    write_inherited_coords: bool = False,
+    last_updated_at: datetime | None = None,
+) -> None:
+    """
+    Write an xarray dataset to an Icechunk store.
+
+    Both `icechunk` and `zarr` (v3) must be installed.
+
+    Parameters
+    ----------
+    dt: xr.DataTree
+        DataTree to write to an Icechunk store. All variables must be backed by
+        ManifestArray objects.
+    store: IcechunkStore
+        Store to write the dataset to, which must not be read-only.
+    write_inherited_coords : bool, default: False
+        If ``True``, replicate inherited coordinates on all descendant nodes of the
+        tree. Otherwise, only write coordinates at the level at which they are
+        originally defined. This saves disk space, but requires opening the
+        full tree to load inherited coordinates.
+    last_updated_at: datetime, optional
+        The time at which the virtual dataset was last updated. When specified, if any
+        of the virtual chunks written in this session are modified in storage after this
+        time, icechunk will raise an error at runtime when trying to read the virtual
+        chunk. When not specified, icechunk will not check for modifications to the
+        virtual chunks at runtime.
+
+    Raises
+    ------
+    ValueError
+        If the store is read-only.
+    """
+    try:
+        from icechunk import IcechunkStore  # type: ignore[import-not-found]
+        from zarr import Group  # type: ignore[import-untyped]
+        from zarr.storage import StorePath  # type: ignore[import-untyped]
+    except ImportError:
+        raise ImportError(
+            "The 'icechunk' and 'zarr' version 3 libraries are required to use this function"
+        ) from None
+
+    if not isinstance(store, IcechunkStore):
+        raise TypeError(
+            f"store: expected type IcechunkStore, but got type {type(store)}"
+        )
+
+    if not isinstance(last_updated_at, (type(None), datetime)):
+        raise TypeError(
+            "last_updated_at: expected type datetime,"
+            f" but got type {type(last_updated_at)}"
+        )
+
+    if store.read_only:
+        raise ValueError("supplied store is read-only")
+
+    for path, subtree in dt.subtree_with_keys:
+        tree = cast(xr.DataTree, subtree)  # subtree is typed as Unknown
+        at_root = tree is dt
+        ds = tree.to_dataset(write_inherited_coords or at_root)
+        store_path = StorePath(store, path="" if at_root else tree.relative_to(dt))
+        group = Group.from_store(store=store_path, zarr_format=3)
+
+        if len(ds):
```

Comment:
I see why you put this here (to only write if there are any variables), but I don't think we want this check. 

Instead, our write function should be able to write to a group even if there are no variables, because even just `ds.attrs` should be written to the store (as zarr group metadata). This may be an edge case that doesn't currently work for `Dataset.virtualize.to_icechunk()`...

Reply:
I removed the length check, and tests still passed. I also tweaked the test to nest 1 of the 2 virtual datasets in the datatree, just to be sure nesting below the root works, and all is good with that too.

IIRC, when I initially started on these changes, I encountered an error when attempting to write an "empty" dataset, which is why I originally added the length check. It seems that my subsequent changes addressed the cause of that problem, so removing the check now works.